### PR TITLE
Modified for Python 3 compatibility

### PR DESCRIPTION
**Pull request by lizihan021**
I left three TODO tags in the code. Please check my modifications around those tags are reasonable.

### CODE REVIEW

File: berkeley_parse_analyser/transform_search.py [hunk 1]
```diff
@@ -390,11 +395,13 @@ def main(args, classify):
 	out_dict['test_trees'] = open(prefix + '.test_trees', 'w')
 	out_dict['error_counts'] = open(prefix + '.error_counts', 'w')
 	out_dict['init_errors'] = open(prefix + '.init_errors', 'w')
-	init.header(args, out_dict.values())
+	# TODO check using list() here is correct.
+	# I did this because otherwise checking the type would be complex
+	init.header(args, list(out_dict.values()))
```
Comment (jkkummerfeld) [inline berkeley_parse_analyser/transform_search.py]: `list(...)` is good here.

File: berkeley_parse_analyser/nlp_util/parse_errors.py [hunk 2]
```diff
@@ -70,7 +71,8 @@ def get_errors(test, gold, include_terminals=False):
         test_spans.remove(span)
 
     gold_spans = [(span.span[0], span.span[1], span) for span in gold]
-    gold_spans.sort()
+    # TODO: check that this change make sense
+    gold_spans.sort(key=lambda x: (x[0], x[1]))
```
[the remark this answers is not in the document]
Comment (lizihan021) [reply berkeley_parse_analyser/nlp_util/parse_errors.py]: I found that if I don't specify the key, the program will throw "Cannot compare object of type "PSTree" and "PSTree" " in Python 3. Maybe Python 3 tries to compare the third element in the tuple, which will generate error. If what we want is to sort the tuple according to the first two elements, I think it is better to specify the keys.

File: berkeley_parse_analyser/nlp_util/pstree.py [hunk 3]
```diff
@@ -329,7 +333,7 @@ def clone_and_find(nodes):
     '''Clone the tree these nodes are in and finds the equivalent nodes in the
     new tree.'''
     return_list = True
-    if not isinstance(nodes, list):
+    if not isinstance(nodes, (list,)):
```
[the remark this answers is not in the document]
Comment (jkkummerfeld) [reply berkeley_parse_analyser/nlp_util/pstree.py]: Hm, I am not finding that in small tests locally. Also, the documentation sounds like a tuple should be optional:

https://docs.python.org/3/library/functions.html#isinstance

[the remark this answers is not in the document]
Comment (lizihan021) [reply berkeley_parse_analyser/nlp_util/pstree.py]: Yes, you are right. Sorry, I should have checked that before making a conclusion. I've changed it back.

File: berkeley_parse_analyser/nlp_util/parse_errors.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jkkummerfeld) [reply berkeley_parse_analyser/nlp_util/parse_errors.py]: Okay, thanks for the fix! These TODOs can be removed - the behaviour should be correct as written here.

File: berkeley_parse_analyser/nlp_util/pstree.py [hunk 4]
```diff
@@ -12,7 +12,7 @@ class TreeIterator:
     PSTree uses pre-order traversal by default, but this supports post-order too, e.g.:
     >>> tree = tree_from_text("(ROOT (S (NP-SBJ (NNP Ms.) (NNP Haag) ) (VP (VBZ plays) (NP (NNP Elianti) )) (. .) ))")
     >>> for node in TreeIterator(tree, 'post'):
-    ...   print node
+    ...   print( node )
```
Comment (jkkummerfeld) [inline berkeley_parse_analyser/nlp_util/pstree.py]: Spacing

**Comment by jkkummerfeld** [comment]
This is looking good! I left a couple of comments that should be quick fixes. Once they are done I'll merge it in.

Note - I did change the pull request to be merging into a new 'development' branch. When I do update master I want to edit the documentation to note the changes (but I'll take care of that).

**Comment by lizihan021** [comment]
I fixed those comments.

**Comment by jkkummerfeld** [comment]
Thanks!

**Pull request merged**